lib/hyperclient/link.rb

```diff
@@ -167,9 +167,17 @@ def _uri_template
 
     def http_method(method, body = nil)
       @resource = begin
-        response = Futuroscope::Future.new do
-          @entry_point.connection.run_request(method, _url, body, nil)
-        end
+        do_async = @entry_point.async_options[:enabled]
```

Comment:
Minor, but this isn't used in more than one place, so I would just do `if @entry_point.async ...`.